jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerMetadata.java

```diff
@@ -33,19 +32,18 @@
   /** Entry into the layer metadata. */
   static class LayerMetadataEntry {
 
-    private List<String> sourceFilesStrings;
+    private ImmutableList<String> sourceFilesStrings;
```

Comment:
Why do these return types need to change?

Reply:
I'm trying to keep everything that is not user-facing (user as in a potential future user of jib-core library) immutable.